src/test/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryServiceTest.java

```diff
@@ -41,301 +40,398 @@ public void setUp() {
         service = new OffenderTransactionHistoryService("GBP", repository);
     }
 
-    @Test
-    public void testGetTransactionHistory_CallsRepositoryWithCorrectParameters() {
+    @Nested
+    public class CorrectParameters {
+        @Test
+        public void testGetTransactionHistoryThrowException_offenderIdIsNull() {
+
+            Throwable exception = assertThrows(NullPointerException.class, () -> {
```

Comment:
AssertJ's `assertThatThrownBy` is worth considering in future - has some of the usual AssertJ goodness.